packages/model-fragments/lib/fragments/attributes.js

```diff
@@ -52,20 +53,19 @@ function hasOneFragment(declaredTypeName, options) {
     options: options
   };
 
-  return Ember.computed(function(key, value) {
-    var record = this;
-    var store = this.store;
-    var data = this._data[key] || getDefaultValue(this, options, 'object');
-    var fragment = this._fragments[key];
-    var actualTypeName = getActualFragmentType(declaredTypeName, options, data);
+  function setOwner(fragment, record, key) {
+    Ember.assert("Fragments can only belong to one owner, try copying instead", !get(fragment, '_owner') || get(fragment, '_owner') === record);
+    return fragment.setProperties({
+      _owner : record,
+      _name  : key
+    });
+  }
```

Comment:
Since this function is no longer using `record` and `key` as closures, how about pulling it out to the top level module scope and renaming it `setFragmentOwner`?